test/NuGet.Clients.Tests/NuGet.VisualStudio.Internal.Contracts.Test/Formatters/IInstalledAndTransitivePackagesFormatterTests.cs

```diff
@@ -0,0 +1,82 @@
+        private static readonly NuGetFramework TransitiveFramework1 = NuGetFramework.Parse("net50");
+        private static readonly IPackageReferenceContextInfo TransitivePackageReferenceContextInfo1 = PackageReferenceContextInfo.Create(TransitivePackageIdentity1, TransitiveFramework1);
+        private static readonly PackageIdentity TransitivePackageIdentity2 = new PackageIdentity(id: "dd", NuGetVersion.Parse("1.2.6"));
+        private static readonly NuGetFramework TransitiveFramework2 = NuGetFramework.Parse("net50");
+        private static readonly IPackageReferenceContextInfo TransitivePackageReferenceContextInfo2 = PackageReferenceContextInfo.Create(TransitivePackageIdentity2, TransitiveFramework2);
+        private static readonly PackageIdentity TransitivePackageIdentity3 = new PackageIdentity(id: "ee", NuGetVersion.Parse("1.2.7"));
+        private static readonly NuGetFramework TransitiveFramework3 = NuGetFramework.Parse("net50");
+        private static readonly IPackageReferenceContextInfo TransitivePackageReferenceContextInfo3 = PackageReferenceContextInfo.Create(TransitivePackageIdentity3, TransitiveFramework3);
+        private static readonly List<IPackageReferenceContextInfo> TransitivePackages = new List<IPackageReferenceContextInfo>() { TransitivePackageReferenceContextInfo1, TransitivePackageReferenceContextInfo2, TransitivePackageReferenceContextInfo3 };
+
+        [Theory]
+        [MemberData(nameof(IInstalledAndTransitivePackages))]
+        public void SerializeThenDeserialize_WithValidArguments_RoundTrips(InstalledAndTransitivePackages expectedResult)
+        {
+            IInstalledAndTransitivePackages actualResult = SerializeThenDeserialize(IInstalledAndTransitivePackagesFormatter.Instance, expectedResult);
+
+            Assert.Equal(actualResult.InstalledPackages.Count, expectedResult.InstalledPackages.Count);
+            Assert.Equal(actualResult.TransitivePackages.Count, expectedResult.TransitivePackages.Count);
+
+            foreach (IPackageReferenceContextInfo expectedPackage in expectedResult.InstalledPackages)
+            {
+                IPackageReferenceContextInfo actualPackage = actualResult.InstalledPackages.FirstOrDefault(p => p.Identity.Equals(expectedPackage.Identity));
+                Assert.NotNull(actualPackage);
+                if (!(actualPackage is null))
+                {
+                    Assert.Equal(expectedPackage.Identity, actualPackage.Identity);
+                    Assert.Equal(expectedPackage.Framework, actualPackage.Framework);
+                    Assert.Equal(expectedPackage.AllowedVersions, actualPackage.AllowedVersions);
+                    Assert.Equal(expectedPackage.IsAutoReferenced, actualPackage.IsAutoReferenced);
+                    Assert.Equal(expectedPackage.IsUserInstalled, actualPackage.IsUserInstalled);
+                    Assert.Equal(expectedPackage.IsDevelopmentDependency, actualPackage.IsDevelopmentDependency);
```

Comment:
Extract this into a `private static void` helper method and call it here and below.

Reply:
Fixed